metaswitch/common/test/test_alarms.py

```diff
@@ -0,0 +1,302 @@
+                                      CLEARED,
+                                      _AlarmManager)
+
+
+class TestAlarmManager(_AlarmManager):
+    """AlarmManager with extra test function."""
+
+    def wake_up(self):
+        """Force the alarm manager to check whether to re-sync."""
+        with self._condition:
+            self._condition.notify()
+
+    def safe_terminate(self):
+        """Terminate on another thread to avoid hanging."""
+        terminate_thread = threading.Thread(target=self.terminate)
+        terminate_thread.daemon = True
+        terminate_thread.start()
+        terminate_thread.join(5)
+
+
+class TestAlarmState(unittest.TestCase):
+    @mock.patch('metaswitch.common.alarms._sendrequest')
+    def test_issue_alarm(self, mock_sendrequest):
+        """Alarm states can be issued."""
+        alarm_state = AlarmState('TestIssuer', 1000, 6)
+        alarm_state.issue()
+        mock_sendrequest.assert_called_once_with(['issue-alarm',
+                                                  'TestIssuer',
+                                                  '1000.6'])
+
+
+class TestBaseAlarm(unittest.TestCase):
+    @mock.patch('metaswitch.common.alarms._sendrequest')
+    def test_clear_base_alarm(self, mock_sendrequest):
+        """Alarms can be cleared."""
+        base_alarm = BaseAlarm('TestIssuer', 1000)
+        base_alarm.clear()
+        mock_sendrequest.assert_called_once_with(['issue-alarm',
+                                                  'TestIssuer',
+                                                  '1000.1'])
+
+
+class TestAlarm(unittest.TestCase):
+    @mock.patch('metaswitch.common.alarms._sendrequest')
+    def test_set_alarm(self, mock_sendrequest):
+        """Alarms can be set."""
+        alarm = Alarm('TestIssuer', 1000, 3)
+        alarm.set()
+        mock_sendrequest.assert_called_once_with(['issue-alarm',
+                                                  'TestIssuer',
+                                                  '1000.3'])
+
+
+    @mock.patch('metaswitch.common.alarms._sendrequest')
+    def test_clear_alarm(self, mock_sendrequest):
+        """Alarm can be cleared after they are set."""
+        alarm = Alarm('TestIssuer', 1000, 3)
+        alarm.set()
+        alarm.clear()
+        mock_sendrequest.assert_called_with(['issue-alarm',
+                                             'TestIssuer',
+                                             '1000.1'])
+
+
+class TestMultiSeverityAlarm(unittest.TestCase):
+    @mock.patch('metaswitch.common.alarms._sendrequest')
+    def test_set_multi_severity_alarm(self, mock_sendrequest):
+        """Multiple severities can be raised."""
+        alarm = MultiSeverityAlarm('TestIssuer', 1000, [3, 6])
+        alarm.set(3)
+        mock_sendrequest.assert_called_once_with(['issue-alarm',
+                                                  'TestIssuer',
+                                                  '1000.3'])
+        mock_sendrequest.reset_mock()
```

Comment:
[reset_mock](http://www.voidspace.org.uk/python/mock/mock.html#mock.Mock.reset_mock) resets the list of calls that the mock thinks have been made to it, so I can check that exactly the right calls have been made (that's why it's a method on the individual mock object). It has no wider effect, so I don't think another test adds anything. Agree?